pwiz_tools/Skyline/Model/Prosit/Models/PrositIntensityModel.cs

```diff
@@ -140,25 +152,36 @@ public override PrositMS2Spectra CreateSkylineOutput(SrmSettings settings, IList
 
         public class PeptidePrecursorNCE : SkylineInputRow
         {
-            public PeptidePrecursorNCE(PeptideDocNode nodePep, TransitionGroupDocNode nodeGroup) : this(nodePep, nodeGroup, nodeGroup.LabelType, null)
+
+            public PeptidePrecursorNCE(string sequence, int precursorCharge, SignedMz precursorMz, ExplicitMods explicitMods, IsotopeLabelType isotopeLabelType, int? nce = null)
             {
+                Sequence = sequence;
+                PrecursorCharge = precursorCharge;
+                PrecursorMz = precursorMz;
+                ExplicitMods = explicitMods;
+                LabelType = isotopeLabelType;
+                NCE = nce;
             }
-            public PeptidePrecursorNCE(PeptideDocNode nodePep, TransitionGroupDocNode nodeGroup, IsotopeLabelType labelType, int? nce)
+
+            public PeptidePrecursorNCE(PeptideDocNode nodePep, TransitionGroupDocNode nodeGroup, IsotopeLabelType isotopeLabelType = null, int? nce = null)
+                : this(nodePep.Peptide.Sequence, nodeGroup.PrecursorCharge, nodeGroup.PrecursorMz, nodePep.ExplicitMods, isotopeLabelType ?? nodeGroup.LabelType, nce)
             {
                 NodePep = nodePep;
                 NodeGroup = nodeGroup;
-                LabelType = labelType;
-                NCE = nce;
             }
 
-            public PeptideDocNode NodePep { get; private set; }
-            public TransitionGroupDocNode NodeGroup { get; private set; }
-            public IsotopeLabelType LabelType { get; private set; }
+            public PeptideDocNode NodePep { get; }
+            public TransitionGroupDocNode NodeGroup { get; }
+            public string Sequence { get; }
+            public int PrecursorCharge { get; }
+            public SignedMz PrecursorMz { get; }
+            public ExplicitMods ExplicitMods { get; }
+            public IsotopeLabelType LabelType { get; }
             public int? NCE { get; private set; }
 
             public PeptidePrecursorNCE WithNCE(int nce)
             {
-                return new PeptidePrecursorNCE(NodePep, NodeGroup, LabelType, nce);
+                return new PeptidePrecursorNCE(Sequence, PrecursorCharge, PrecursorMz, ExplicitMods, LabelType, NCE);

```

Comment:
This "NCE" in capitals suspicious. "PeptidePrecursorNCE" be constructed with the "nce" parameter that was passed in, instead of the NCE property?
Also, "CreatePrositInputRow" definitely wants a non-null PeptideDocNode, which causes "TestPrositBuildLibrary" to fail.

Reply:
That was definitely suspicious. I think I fixed these issues.